### PR TITLE
Fix registration

### PR DESCRIPTION
**Pull request by giomfo**

### CODE REVIEW

File: MatrixKit/Controllers/MXKAuthenticationViewController.m [hunk 1]
```diff
@@ -89,6 +89,39 @@ + (instancetype)authenticationViewController
                                           bundle:[NSBundle bundleForClass:[MXKAuthenticationViewController class]]];
 }
 
+#pragma mark -
+
+- (instancetype)initWithNibName:(nullable NSString *)nibNameOrNil bundle:(nullable NSBundle *)nibBundleOrNil
```
Comment (manuroe) [inline MatrixKit/Controllers/MXKAuthenticationViewController.m]: We should really add these methods to top parent (MXKViewController and MXKTableViewController).

File: MatrixKit/Controllers/MXKAuthenticationViewController.m [hunk 2]
```diff
@@ -621,6 +670,61 @@ - (void)handleAuthenticationSession:(MXAuthenticationSession *)authSession
...
+        // Remove the current auth inputs view
+        self.authInputsView = nil;
+        
+        // Set external parameters and trigger a refresh (the parameters will be taken into account during [handleAuthenticationSession:])
+        _externalRegistrationParameters = parameters;
+        [self refreshAuthenticationSession];
+    }
+    else
+    {
+        NSLog(@"[MXKAuthenticationVC] resetExternalRegistrationParameters");
```
Comment (manuroe) [inline MatrixKit/Controllers/MXKAuthenticationViewController.m]: It may worth to use a different log string than in the "then" block.

**Comment by manuroe** [comment]
Except the minor comment on the NSLog, LVGTM

File: MatrixKit/Controllers/MXKAuthenticationViewController.m (hunk 1 above)
Comment (giomfo) [inline MatrixKit/Controllers/MXKAuthenticationViewController.m]: An issue has been created: https://github.com/matrix-org/matrix-ios-kit/issues/94

**Pull request merged**